### PR TITLE
Enable sriov support by default

### PR DESCRIPTION
**Pull request by fedepaol**
**What this PR does / why we need it**:

This pr makes the sr-iov support enabled by default by removing the need for feature gate. 

**Release note**:

```release-note

SR-IOV support enabled by default, removed feature gate.

```

### CODE REVIEW

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go [hunk 1]
```diff
@@ -1735,7 +1735,7 @@ var _ = Describe("Validating VMICreate Admitter", func() {
 			Expect(len(causes)).To(Equal(0))
 		})
 
-		It("should reject SRIOV interface when feature gate is disabled", func() {
+		It("should accept SRIOV interface when feature gate is disabled", func() {
```
Comment (booxter) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go]: In my mind, we have no code for SR-IOV in the admitter so it's arguable if we need to have any test cases for sriov here. But I am ok either way.

**Comment by fedepaol** [comment]
> I'd like to better understand why we introduced the feature gate, and why we remove it now. Was this discussed somewhere?

My understanding is that sriov support was kind of experimental and we did not have proper ci around. Also, it's opt-in per se since it won't get activated if a sriov card is not explicitly added to a guest, but I'd also wait for @booxter if he has more info about this

**Comment by booxter** [comment]
@slintes there were two main reasons: 1) no upstream CI for the feature; and 2) SR-IOV interface attached enabled privileged mode for the pod. Both issues are now tackled, hence now we enable it by default.

**Comment by slintes** [comment]
@fedepaol ack, thanks! Works for me.
@rmohr is this ok for you too?
@fedepaol the user-guide needs an update as well.

/lgtm

**Comment by fedepaol** [comment]
> @booxter ack, thanks! Works for me.
> @rmohr is this ok for you too?
> @fedepaol the user-guide needs an update as well.

Will do

> 
> /lgtm

**Comment by fedepaol** [comment]
User guide pr: https://github.com/kubevirt/user-guide/pull/270

**Comment by rmohr** [comment]
> @rmohr is this ok for you too?

Since CI is now in place, no issue with it. :+1:

**Pull request merged**